website/project/model.py

```diff
@@ -1149,6 +1177,31 @@ def has_active_draft_registrations(self):
         else:
             return True
 
+    @property
+    def is_preprint(self):
+        if not self.preprint_file:
+            return False
+        if self.preprint_file.node == self:
+            return True
+        else:
+            self.preprint_file = None
+            self._is_preprint_orphan = True
```

Comment:
Unless you save after checking `.is_preprint`, these field changes won't persist. You don't typically want to be calling `self.save` inside of a property, but this might be an exception to that rule.